backend/src/gpu.py

```diff
@@ -0,0 +1,51 @@
+import pynvml as nv
+from sanic.log import logger
+
+# pylint: disable=global-at-module-level
+global nvidia_is_available
+nvidia_is_available = False
+
+try:
+    nv.nvmlInit()
+    nvidia_is_available = True
+    nv.nvmlShutdown()
+except nv.NVMLError as e:
+    logger.info("No Nvidia GPU found, or invalid driver installed.")
+except Exception as e:
+    logger.info(f"Unknown error occurred when trying to initialize Nvidia GPU: {e}")
+
+
+class NvidiaHelper:
+    def __init__(self):
+        nv.nvmlInit()
+
+        self.__num_gpus = nv.nvmlDeviceGetCount()
+
+        self.__gpus = []
+        for i in range(self.__num_gpus):
+            handle = nv.nvmlDeviceGetHandleByIndex(i)
+            self.__gpus.append(
+                {
+                    "name": nv.nvmlDeviceGetName(handle),
+                    "uuid": nv.nvmlDeviceGetUUID(handle),
+                    "index": i,
+                    "handle": handle,
+                }
+            )
+
+    def __del__(self):
+        nv.nvmlShutdown()
+
+    def list_gpus(self):
+        return self.__gpus
+
+    def get_current_vram_usage(self, gpu_index=0):
+        info = nv.nvmlDeviceGetMemoryInfo(self.__gpus[gpu_index]["handle"])
+
+        return info.total, info.used, info.free
+
+
+__all__ = [
+    "nvidia_is_available",
+    "NvidiaHelper",
+]
```

Comment:
Do we need this?

Reply:
not really, but i kinda think this is something we should start doing to avoid leaking imported things as exports. python's default implementation is "export everything" which in the past has caused us to use things in one file that we import via importing something else (which while that might be ok, I personally hate). Using this is like how nodejs does explicit exports. Without this, it would be exporting sanic's logger and the pynvml package

Reply:
alright. It's good as is then.